udplog/syslog.py

```diff
@@ -0,0 +1,184 @@
+# -*- test-case-name: udplog.test.test_syslog -*-
+#
+# Copyright (c) Ralph Meijer.
+# See LICENSE for details.
+
+"""
+Syslog server support.
+
+This provides support for receiving syslog messages and utilities for
+further shipping similar to the native udplog protocol.
+"""
+
+from __future__ import division, absolute_import
+
+import calendar
+import json
+import re
+
+from dateutil.parser import parse
+from dateutil import tz
+
+from twisted.internet.protocol import DatagramProtocol
+from twisted.python import log
+
+FACILITIES = [
```

Comment:
Unfortunately, it doesn't seem that way. The standard library has a [syslog](https://docs.python.org/2/library/syslog.html) module that defines [constants](http://svn.python.org/projects/python/trunk/Modules/syslogmodule.c) for the facility and severity levels, but unfortunately doesn't provide some kind of ordered collection of them, and no reverse mapping. Similarly, there is a standard library logging handled [logging.handlers.SysLogHandler](https://docs.python.org/2/library/logging.handlers.html#sysloghandler) which defines a function to go from a name to a value, but not the reverse. The internal dict that holds that mapping doesn't really appear to be public.